maven-core/src/main/java/org/apache/maven/classrealm/DefaultClassRealmManager.java

```diff
@@ -301,20 +305,22 @@ private void callDelegates(
     }
 
     private Set<String> populateRealm(ClassRealm classRealm, List<ClassRealmConstituent> constituents) {
-        Set<String> includedIds = new LinkedHashSet<>();
+        Set<String> includedIds = Collections.emptySet();
```

Comment:
What is your concerns? The **includedIds** reference a collection that stays local to the methods **populateRealm** and **createRealm**, it is not escaping the class and is not an instance variable. The set is only modified if the logger debug log level is enabled (line 319 and 321), it is set with a LinkedHashSet on the same condition (lines 310 and 311). I see no point of creating a LinkedHashSet if the log level is not at least debug since this collection is only used for logging purpose. What do you think according to this?

Reply:
You're saying that at this commit, mixing mutable and immutable lists is not a problem. That's likely true. However there's no guarantee this will still be true in a commit 5 years down the road. I've seen slipping an immutable list into something that wasn't expecting it cause very painful production problems. It's just safer to always use one or the other, and make sure the declared types make it clear which is being used. Any collection not explicitly declared as immutable should not be immutable. Yes, this is a design flaw in the Java Collections API going back to Java 1.1, but it's a design flaw we have to live with.

Reply:
Fine with me, thanks for clarifying. Reverting this line